src/openforms/analytics_tools/models.py

```diff
@@ -0,0 +1,308 @@
+            return super().save(*args, **kwargs)
+        original_object = self.__class__.objects.get(pk=self.pk)
+        # For each analytics provider, we check if :
+        # - the value of the provider's boolean field is changing
+        if original_object.enable_google_analytics != self.enable_google_analytics:
+            self.update_analytics_tool(
+                AnalyticsTools.google_analytics, self.is_google_analytics_enabled, []
+            )
+        if (
+            original_object.enable_matomo_site_analytics
+            != self.enable_matomo_site_analytics
+        ):
+            string_replacements_list = [
+                ("SITE_ID", self.matomo_site_id),
+                ("SITE_URL", self.matomo_url),
+                (
+                    "DOMAIN_HASH",
+                    lambda x: get_domain_hash(settings.ALLOWED_HOSTS[0], cookie_path=x),
+                ),
+            ]
+            self.update_analytics_tool(
+                AnalyticsTools.matomo, self.is_matomo_enabled, string_replacements_list
+            )
+        if (
+            original_object.enable_piwik_pro_site_analytics
+            != self.enable_piwik_pro_site_analytics
+        ):
+
+            string_replacements_list = [
+                ("SITE_ID", self.piwik_pro_site_id),
+                ("SITE_URL", self.piwik_pro_url),
+                (
+                    "DOMAIN_HASH",
+                    lambda x: get_domain_hash(settings.ALLOWED_HOSTS[0], cookie_path=x),
+                ),
+            ]
+            self.update_analytics_tool(
+                AnalyticsTools.piwik_pro,
+                self.is_piwik_pro_enabled,
+                string_replacements_list,
+            )
+        if (
+            original_object.enable_piwik_site_analytics
+            != self.enable_piwik_site_analytics
+        ):
+
+            string_replacements_list = [
+                ("SITE_ID", self.piwik_site_id),
+                ("SITE_URL", self.piwik_url),
+                (
+                    "DOMAIN_HASH",
+                    lambda x: get_domain_hash(settings.ALLOWED_HOSTS[0], cookie_path=x),
+                ),
+            ]
+
+            self.update_analytics_tool(
+                AnalyticsTools.piwik, self.is_piwik_enabled, string_replacements_list
+            )
+        if (
+            original_object.enable_siteimprove_analytics
+            != self.enable_siteimprove_analytics
+        ):
+            self.update_analytics_tool(
+                AnalyticsTools.siteimprove, self.is_siteimprove_enabled, []
+            )
+        super().save(*args, **kwargs)
+
+    def clean(self):
+        if self.enable_google_analytics and not self.is_google_analytics_enabled:
+            raise ValidationError(
+                _(
+                    "If you enable {analytics_tool}, you must fill out all the required fields".format(
+                        analytics_tool="Google Analytics"
+                    )
+                )
```

Comment:
```suggestion
                _("If you enable {analytics_tool}, you must fill out all the required fields").format(
                    analytics_tool="Google Analytics"
                )
```
this is a common mistake - you want to apply the `.format` on the string marked for translation, not mark the resulting interpolated string translatable.